cmake/modules/BuildOpentelemetry.cmake

```diff
@@ -82,4 +82,5 @@ function(build_opentelemetry)
     PROPERTIES
       INTERFACE_LINK_LIBRARIES "${opentelemetry_deps}"
       INTERFACE_INCLUDE_DIRECTORIES "${opentelemetry_include_dir}")
+  include_directories(SYSTEM "${opentelemetry_include_dir}")
```

Comment:
nit: since we have the `opentelemetry::libopentelemetry` target for this header dependency, we should probably require targets like ceph-common to depend on that target or `jaeger_base` instead of adding a global `include_directories()`

Reply:
do you mean I should add the include dir and dependency separatly to each Cmake that needs that?
i.e. add:
```cmake
if(WITH_JAEGER)
  add_dependencies(<target> opentelemetry::libopentelemetry)
  target_include_directories(<targte> PRIVATE 
    $<TARGET_PROPERTY:opentelemetry::libopentelemetry,INTERFACE_INCLUDE_DIRECTORIES>)
endif()
```
to: common, msg, auth, global, librados etc.

Reply:
ideally we'd just add `target_link_libraries(common PUBLIC opentelemetry::libopentelemetry)` and other targets would inherit the dependency from that

but i recently went through a similar exercise trying to remove the [global include_directories()](https://github.com/ceph/ceph/blob/370e0ed/src/CMakeLists.txt#L326-L328) for libfmt in https://github.com/ceph/ceph/pull/50479, and it was a huge mess because of all of the separate object libraries like auth, msg, etc that get linked into common

so let's leave this as-is, and think about cleaning up that mess of common dependencies later.. maybe.. some day